summarizer/main.py

```diff
@@ -95,3 +109,6 @@ def test(hps):
         test(hps)
     else:
         train(hps)
+
+    # Close the Tensorboard writer
+    hps.writer.close()
```

Comment:
Suggestion: transfer this into config class within the `def __del__(self):`

Reply:
Update: I tried but it failed, when `self.writer.close()` is called in the `__del__(self)` it no longer works (wait infinitely). After further investigation, the `.close()` on the writers in the loop [here](https://github.com/pytorch/pytorch/blob/3e3498cf03b6ea7becf34597eea7ff350d718f95/torch/utils/tensorboard/writer.py#L1019) fails

Reply:
This needs more investigation. I see some workarounds here: https://github.com/facebookresearch/detectron2/issues/121

Reply:
It seems like workarounds for this revolve around external/calling code closing the writer. While not optimal, this is how we have it implemented currently as well.